contents/transform.md

```diff
@@ -0,0 +1,94 @@
+## Variable Transformations
+
+```{=comment}
+Ifelse and case_when
+```
+
+In @sec:filter, we saw that `filter` works by taking one or more source columns and filtering it by applying a function.
+In other words, we saw that `source => f::Function` like in, for example, `filter(:name => name -> name == "Alice", df)`.
+In @sec:select, we saw that `select` can take one or more source columns and put it into one or more target columns `source => target` like in, for example, `select(df, :name => :people_names)`.
+In this section, we discuss how to **transform** variables, that is, how to modify data.
+This goes via `source => transformation => target`.
+
+As a generous example, given the `grades_2020` dataset:
+
+```jl
+@sco process=without_caption_label grades_2020()
+```
+
+we can increase all the grades in `grades_2020` by 1:
+
+```jl
+s = """
+    plus_one(grades) = grades .+ 1
+    transform(grades_2020(), :grade_2020 => plus_one)
+    """
+sco(s; process=without_caption_label)
+```
+
+Here, the `plus_one` function receives the whole `:grade_2020` row and not just one element as one might expect.
+That is the reason why we've added the dot before the plus symbol.
+The `dot`, in Julia's syntax, means broadcasting and allows Julia to add one to all the elements in `grades`.
+For instance:
+
+```jl
+sco("[1, 2, 3] .+ 1")
+```
+
+Like said above, the `DataFrames.jl` minilanguage is `source => transformation => target`.
+So, if we want to have only one column in the output, we can do:
```

Comment:
```suggestion
So, if we want to keep the naming of the `target` column in the output, we can do:
```


I think it's easier to say that if "we want to keep the naming of the target column".

Because the example before without the `=> target` works fine also.

Also you'll probably need to cover

```julia
transform(df, ...; renamecols=false)
```

which I use a lot.